src/leveled_iclerk.erl

```diff
@@ -549,44 +584,31 @@ safely_log_filescore(PositionList, FN, Score, SW) ->
     leveled_log:log_timer("IC004", [Score, AvgJump, FN], SW).
 
 -spec size_comparison_score(list(key_size() | corrupted_test_key_size()),
-                                    fun(),
-                                    any(),
-                                    non_neg_integer(),
+                                    leveled_inker:filterfun(),
+                                    leveled_inker:filterserver(),
+                                    leveled_codec:sqn(),
                                     leveled_codec:compaction_strategy()) ->
                                         float().
 size_comparison_score(KeySizeList,
                         FilterFun, FilterServer, MaxSQN,
-                        RS) ->
+                        ReloadStrategy) ->
     FoldFunForSizeCompare =
         fun(KS, {ActSize, RplSize}) ->
             case KS of
                 {{SQN, Type, PK}, Size} ->
-                    IsJournalEntry =
-                        leveled_codec:is_full_journalentry({SQN, Type, PK}),
-                    case IsJournalEntry of
-                        false ->
-                            TS = leveled_codec:get_tagstrategy(PK, RS),
-                            % If the strategy is to retain key deltas, then
-                            % scoring must reflect that.  Key deltas are
-                            % possible even if strategy does not allow as
-                            % there is support for changing strategy from
-                            % retain to recalc
-                            case TS of
-                                retain ->
-                                    {ActSize + Size - ?CRC_SIZE, RplSize};
-                                _ ->
-                                    {ActSize, RplSize + Size - ?CRC_SIZE}
-                            end;
+                    ToRetain =
+                        to_retain({SQN, Type, PK},
+                                    FilterFun,
+                                    FilterServer,
+                                    MaxSQN,
+                                    ReloadStrategy),
+                    case ToRetain of
                         true ->
-                            Check = FilterFun(FilterServer, PK, SQN),
-                            case {Check, SQN > MaxSQN} of
-                                {current, _} ->
```

Comment:
the old code can take this path without strategy being "retain" but new code cannot.